joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: &loop_hz 500

ur10e_eff_joint_traj_controller:
  type: effort_controllers/JointTrajectoryController
  joints: &robot_joints
    - ur10e_shoulder_pan_joint
    - ur10e_shoulder_lift_joint
    - ur10e_elbow_joint
    - ur10e_wrist_1_joint
    - ur10e_wrist_2_joint
    - ur10e_wrist_3_joint
  gains: # Required because we're controlling an effort interface
    ur10e_shoulder_pan_joint: {p: 2000,  d: 100, i: 10, i_clamp: 1}
    ur10e_shoulder_lift_joint: {p: 20000,  d: 100, i: 20, i_clamp: 1}
    ur10e_elbow_joint: {p: 10000,  d: 100, i: 10, i_clamp: 1}
    ur10e_wrist_1_joint: {p: 500,  d: 1, i: 1, i_clamp: 1}
    ur10e_wrist_2_joint: {p: 500,  d: 1, i: 1, i_clamp: 1}
    ur10e_wrist_3_joint: {p: 10,  d: 0.1, i: 0, i_clamp: 1}  
  constraints:
    goal_time: 0.6
    stopped_velocity_tolerance: 0.01
    ur10e_shoulder_pan_joint: {trajectory: 0.05, goal: 0.02}
    ur10e_shoulder_lift_joint: {trajectory: 0.05, goal: 0.02}
    ur10e_elbow_joint: {trajectory: 0.05, goal: 0.02}
    ur10e_wrist_1_joint: {trajectory: 0.05, goal: 0.02}
    ur10e_wrist_2_joint: {trajectory: 0.05, goal: 0.02}
    ur10e_wrist_3_joint: {trajectory: 0.05, goal: 0.02}
  stop_trajectory_duration: 0.5
  state_publish_rate: *loop_hz
  action_monitor_rate: 10